returnn/tensor/tensor.py

```diff
@@ -180,7 +186,7 @@ def raw_tensor(self, value: Optional[RawTensorType]):
                 if dim.dimension is None:
                     continue  # we allow anything in the raw_tensor dim
                 if raw_shape[i] != dim.dimension:
-                    raise Exception(
+                    raise ShapeMismatchException(
```

Comment:
Undo this change here. This does not belong into this PR. And this is not consistent. There are many other places throughout the `Tensor` code where shapes are checked, where it does not use this exception type consistently now. This requires much more code changes though, and they definitely do not belong here in this PR. And also not sure if really needed.